### PR TITLE
Add deep link support

### PR DESCRIPTION
**Pull request by mehdiyari**
Add deep link support for 
home(http://moallemi.me/home)
dashboard(http://moallemi.me/dashboard)
notifications(http://moallemi.me/notifications)
pages(http://moallemi.me/pages/{page_number})

### CODE REVIEW

File: app/src/main/java/me/moallemi/multinavhost/MainActivity.kt [hunk 1]
```diff
@@ -15,12 +15,11 @@ class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemS
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-
+        if (savedInstanceState == null) tabManager.currentController = tabManager.navHomeController
```
Comment (moallemi) [inline app/src/main/java/me/moallemi/multinavhost/MainActivity.kt]: please do not use single line `if` 

```suggestion
        if (savedInstanceState == null) {
            tabManager.currentController = tabManager.navHomeController
        }
```

File: app/src/main/java/me/moallemi/multinavhost/MainActivity.kt [hunk 2]
```diff
@@ -15,12 +15,11 @@ class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemS
...
         bottomNavigationView.setOnNavigationItemSelectedListener(this)
 
-        if (savedInstanceState == null) {
-            tabManager.currentController = tabManager.navHomeController
-        }
+        if (savedInstanceState == null && intent.action === Intent.ACTION_VIEW && intent?.data != null)
```
Comment (moallemi) [inline app/src/main/java/me/moallemi/multinavhost/MainActivity.kt]: same comment here

File: app/src/main/java/me/moallemi/multinavhost/MainActivity.kt [hunk 3]
```diff
@@ -48,4 +47,30 @@ class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemS
         return true
     }
 
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        setIntent(intent)
+        if (intent != null && intent.action === Intent.ACTION_VIEW && intent.data != null)
+            handleDeepLink()
```
Comment (moallemi) [inline app/src/main/java/me/moallemi/multinavhost/MainActivity.kt:57]: same comment here

**Review by moallemi — CHANGES_REQUESTED** [review]
@mehdiyari Thank you for your great job!

Your changes handle deep links nicely, but there is a small issue:

Go through these steps:
1- open app (now you are on home tab)
2- open notifications tab via deep link:
```
adb shell am start -a android.intent.action.VIEW -d "http://moallemi.me/notifications"
```
3- app navigates to notifications tab. now press back to go back to previous tab
**Actual result:** you have to press back button twice to go back to previous tab (home tab)
**Expected result:** you can navigate back to previous tab only by pressing back button once.

**Review by moallemi — APPROVED** [review]

**Comment by moallemi** [comment]
Resolves #1

**Comment by mehdiyari** [comment]
thanks @moallemi

**Pull request merged**